api/tests/integration/features/workflows/test_integration_workflows.py

```diff
@@ -0,0 +1,121 @@
+    )
+    assert create_cr_response.status_code == status.HTTP_201_CREATED
+    create_cr_response_json = create_cr_response.json()
+    change_request_id = create_cr_response_json["id"]
+
+    # Now, let's verify that nothing has changed with the flags & identities responses
+    for url in (flags_url, identities_url):
+        response = sdk_client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+        flags = response.json() if url == flags_url else response.json()["flags"]
+        assert flags == pre_cr_responses[url]
+
+    # Now, before we approve the CR, let's verify that it's not possible to commit
+    # without approval
+    commit_cr_url = reverse(
+        "api-v1:features:workflows:change-requests-commit", args=(change_request_id,)
+    )
+    invalid_commit_cr_response = admin_client.post(commit_cr_url)
+    assert invalid_commit_cr_response.status_code == status.HTTP_400_BAD_REQUEST
+    assert (
+        invalid_commit_cr_response.json()["detail"]
+        == "Change request has not been approved by all required approvers."
+    )
+
+    # Now, let's approve the CR and commit it
+    approve_cr_url = reverse(
+        "api-v1:features:workflows:change-requests-approve", args=(change_request_id,)
+    )
+    registered_user_client = APIClient()
+    registered_user_client.credentials(
+        HTTP_AUTHORIZATION=f"Token {registered_user_token}"
+    )
+    approve_cr_response = registered_user_client.post(approve_cr_url)
+    assert approve_cr_response.status_code == status.HTTP_200_OK
+
+    commit_cr_response = admin_client.post(commit_cr_url)
+    assert commit_cr_response.status_code == status.HTTP_200_OK
+
+    # Now, let's verify that the flags & identities responses correctly reflect the
+    # change from the CR
+    for url in (flags_url, identities_url):
+        response = sdk_client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+        flags = response.json() if url == flags_url else response.json()["flags"]
+        assert len(flags) == 1
+        assert flags[0]["enabled"] is True  # new value
+        assert flags[0]["feature_state_value"] == "foobar"  # new value
+
+    # Finally, let's test retrieving and updating a change request
+    cr_detail_url = reverse(
+        "api-v1:features:workflows:change-requests-detail", args=(change_request_id,)
+    )
+    retrieve_response = admin_client.get(cr_detail_url)
+    assert retrieve_response.status_code == status.HTTP_200_OK
+
+    # when we update, let's just change the title
+    change_request_json = retrieve_response.json()
+    new_title = "%s (edit)" % change_request_json["title"]
+    change_request_json["title"] = new_title
+    update_response = admin_client.put(
+        cr_detail_url,
+        data=json.dumps(change_request_json),
+        content_type="application/json",
+    )
+    assert update_response.status_code == status.HTTP_200_OK
```

Comment:
Should we assert that `update_response.json()["title"] == new_title`?

Reply:
We definitely should yes, I'll do that now.

Reply:
Added.